.github/workflows/regression-report.yml

```diff
@@ -348,18 +349,21 @@ jobs:
     outputs:
       report_label: ${{ needs.handle_configs_in_pr_description.outputs.report_label }}
     steps:
-      - name: Restore Master Checkstyle Checkout
-        uses: actions/cache@v4
+      - name: Restore cache
+        uses: actions/cache/restore@v4
         with:
           path: .
-          key: "${{ env.CACHE_KEY_MASTER }}"
+          key: ${{ env.CACHE_KEY }}
+          restore-keys: |
+            checkouts-
+          fail-on-cache-miss: true
 
       - name: Restore Generated Config and Project Files
         if: needs.handle_generated_config_bundle.result == 'success'
         uses: actions/cache@v4
         with:
           path: ${{ env.GENERATED_CONFIGS_PATH }}
-          key: ${{ env.CACHE_KEY_GENERATED_CONFIGS }}
```

Comment:
uses: actions/cache/restore


and apply this to all others, to make it clear what we are doing.